docs/dev/relay_bring_your_own_codegen.rst

```diff
@@ -516,10 +516,417 @@ So that users can configure whether to include your compiler when configuring TV
+
+	  PackedFunc GetFunction(const std::string& name,
+	                         const ObjectPtr<Object>& sptr_to_self) final;
+
+	  const char* type_key() const { return "examplejson"; }
+
+	  void SaveToBinary(dmlc::Stream* stream) final;
+
+	  static Module LoadFromBinary(void* strm);
+
+	  static Module Create(const std::string& path);
+
+          std::string GetSource(const std::string& format = "");
+
+          void Run(int id, const std::vector<int>& inputs, int output);
+
+          void ParseJson(const std::string& json);
+
+	 private:
+	  /* \brief The json string that represents a computational graph. */
+	  std::string graph_json_;
+	  /* \brief The subgraph that being processed. */
+	  std::string curr_subgraph_;
+	  /*! \brief A simple graph from subgraph id to node entries. */
+	  std::map<std::string, std::vector<NodeEntry> > graph_;
+	  /* \brief A simple pool to contain the tensor for each node in the graph. */
+	  std::vector<NDArray> data_entry_;
+	  /* \brief A mapping from node id to op name. */
+	  std::vector<std::string> op_id_;
+	};
+
+In particular, there are some functions derived from ``ModuleNode`` that we must implement in ``ExampleJsonModule``:
+
+* Constructor: The constructor of this class should accept a subgraph (in your representation), process and store it in any format you like. The saved subgraph could be used by the following two functions.
+
+* ``GetFunction``: This is the most important function in this class. When TVM runtime wants to execute a subgraph with your compiler tag, TVM runtime invokes this function from your customized runtime module. It provides the function name as well as runtime arguments, and ``GetFunction`` should return a packed function implementation for TVM runtime to execute.
+
+* ``SaveToBinary`` and ``LoadFromBinary``: ``SaveToBinary`` serialize the runtime module to a binary format for later deployment. This function will be called by TVM when users use ``export_library`` API. On the other hand, since we are now using our own graph representation, we have to make sure that ``LoadFromBinary`` is able to construct the same runtime module by taking the serialized binary generated by ``SaveToBinary``.
+
+* ``GetSource`` (optional): If you would like to see the generated ExampleJSON code, you can implement this function to dump it; otherwise you can skip the implementation.
+
+Other functions and class variables will be introduced along with the implementation of above must-have functions.
+
+Implement Constructor
+---------------------
+
+.. code-block:: c++
+
+    explicit ExampleJsonModule(std::string graph_json) {
+      this->graph_json_ = graph_json;
+      ParseJson(this->graph_json_);
+    }
+
+Then, we implement ``ParseJson`` to parse a subgraph in ExampleJSON format and construct a graph in memory for later usage. Since we do not support subgraph with branches in this example, we simply use an array to store every nodes in a subgraph in order.
```

Comment:
Add a summary section to show a checklist of things todo when adding a customized runtime and compiler